File: SearchMavenCentral/src/net/worthington/android/maven/search/restletapi/dao/MavenCentralResponse.java
package net.worthington.android.maven.search.restletapi.dao;

import org.codehaus.jackson.annotate.JsonIgnoreProperties;

//TODO: figure out how to parse highlighting
//http://wiki.fasterxml.com/JacksonFAQ#Deserializing:_unknown_properties
@JsonIgnoreProperties({ "highlighting" })
public class MavenCentralResponse
{
  private MCRResponseHeader iResponseHeader;
  private MCRResponse       iResponse;
  private MCRSpellCheck     iSpellcheck;

  public MCRResponseHeader getResponseHeader()
  {
    return iResponseHeader;
  }

  public void setResponseHeader(MCRResponseHeader responseHeader)
  {
    iResponseHeader = responseHeader;
  }

  public MCRResponse getResponse()
  {
    return iResponse;
  }

  public void setResponse(MCRResponse response)
  {
    iResponse = response;
  }

  public MCRSpellCheck getSpellcheck()
  {
    return iSpellcheck;
  }

  public void setSpellcheck(MCRSpellCheck spellcheck)
  {
    iSpellcheck = spellcheck;
  }
}

File: searchmavenapp/lib/api/mavencentral/model/mavencentralresponse.dart
import 'mcr_response.dart';

//from: ./src/com/searchmavenapp/android/maven/search/restletapi/dao/MavenCentralResponse.java
class MavenCentralResponse {
  final Map responseHeader;
  final MCRResponse response;
  final Map spellcheck;

  MavenCentralResponse({
    required this.responseHeader,
    required this.response,
    required this.spellcheck,
  });

  factory MavenCentralResponse.fromJson(Map<String, dynamic> json) =>
      MavenCentralResponse(
        responseHeader: json['responseHeader'],
        response: MCRResponse.fromJson(json['response']),
        spellcheck: json['spellcheck'],
      );
}
